From afe60db4e7d7c98f354a21434a0515c96f5054c6 Mon Sep 17 00:00:00 2001
From: Marcin Niestroj <m.niestroj@emb.dev>
Date: Sat, 11 Nov 2023 16:03:47 +0100
Subject: [PATCH] samples: http_get: add tests with Native Simulator offloaded
 sockets

'http_get' is one of the few samples that just require connection to
internet (google.com over HTTP or HTTPS) and check if HTTP GET was
successful. Use it for minimal coverage of Native Simulator offloaded
sockets:

 * DNS resolving
 * basic TCP transfer
 * compatibility with Zephyr TLS subsystem (which required some fcntl()
   operations)

Signed-off-by: Marcin Niestroj <m.niestroj@emb.dev>
---
 .../net/sockets/http_get/overlay-nsos.conf    |  6 ++++
 samples/net/sockets/http_get/sample.yaml      | 30 ++++++++++++++++++-
 2 files changed, 35 insertions(+), 1 deletion(-)
 create mode 100644 samples/net/sockets/http_get/overlay-nsos.conf

diff --git a/samples/net/sockets/http_get/overlay-nsos.conf b/samples/net/sockets/http_get/overlay-nsos.conf
new file mode 100644
index 00000000000..0af19c1b112
--- /dev/null
+++ b/samples/net/sockets/http_get/overlay-nsos.conf
@@ -0,0 +1,6 @@
+CONFIG_ETH_NATIVE_POSIX=n
+CONFIG_NET_DRIVERS=y
+CONFIG_NET_SOCKETS=y
+CONFIG_NET_SOCKETS_OFFLOAD=y
+CONFIG_NET_NATIVE_OFFLOADED_SOCKETS=y
+CONFIG_HEAP_MEM_POOL_SIZE=1024
diff --git a/samples/net/sockets/http_get/sample.yaml b/samples/net/sockets/http_get/sample.yaml
index 2de78ce0bee..4bdb915f2e5 100644
--- a/samples/net/sockets/http_get/sample.yaml
+++ b/samples/net/sockets/http_get/sample.yaml
@@ -3,7 +3,6 @@ sample:
   name: socket_http_get
 common:
   filter: CONFIG_FULL_LIBC_SUPPORTED
-  harness: net
   min_ram: 32
   min_flash: 80
   tags:
@@ -11,12 +10,14 @@ common:
     - socket
 tests:
   sample.net.sockets.http_get:
+    harness: net
     extra_configs:
       - CONFIG_NET_SOCKETS_POSIX_NAMES=y
       # Forcibly defines CONFIG_POSIX_API, which is incompatible with
       # CONFIG_NET_SOCKETS_POSIX_NAMES.
   sample.net.sockets.http_get.posix:
     filter: not CONFIG_NET_SOCKETS_OFFLOAD and not CONFIG_NATIVE_LIBC
+    harness: net
     platform_exclude:
       - cc3220sf_launchxl
       - cc3235sf_launchxl
@@ -25,8 +26,35 @@ tests:
       - CONFIG_POSIX_API=y
   sample.net.sockets.http_get.offload.simplelink:
     platform_allow: cc3220sf_launchxl
+    harness: net
     tags:
       - net
       - socket
       - offload
       - simplelink
+  sample.get.sockets.http_get.nsos.http:
+    harness: console
+    harness_config:
+      type: multi_line
+      ordered: true
+      regex:
+        - "(.*)<BODY>(.*)"
+        - "The document has moved"
+        - "(.*)</BODY>(.*)"
+    platform_allow:
+      - native_sim
+      - native_sim/native/64
+    extra_args: OVERLAY_CONFIG="overlay-nsos.conf"
+  sample.get.sockets.http_get.nsos.https:
+    harness: console
+    harness_config:
+      type: multi_line
+      ordered: true
+      regex:
+        - "(.*)<BODY>(.*)"
+        - "The document has moved"
+        - "(.*)</BODY>(.*)"
+    platform_allow:
+      - native_sim
+      - native_sim/native/64
+    extra_args: OVERLAY_CONFIG="overlay-nsos.conf;overlay-tls.conf"
-- 
2.45.2

